LogMonitor/src/LogMonitor/Utility.cpp

```diff
@@ -269,12 +269,16 @@ void Utility::SanitizeJson(_Inout_ std::wstring& str)
     while (i < str.size()) {
         auto sub = str.substr(i, 1);
         if (sub == L"\"") {
-            if ((i > 0 && str.substr(i - 1, 1) != L"\\")
+            if ((i > 0 && str.substr(i - 1, 1) != L"\\" && str.substr(i - 1, 1) != L"~")

```

Comment:
Which scenario would we hit this block? I have tried sanitizing the following string with the logic and I have not hit this block.

```
{~"TimeStamp~":~"2023-04-19T10:25:23.000Z~", ~"source~":~"ETW~", ~"Severity~":~"Information~", ~"ProviderId~":~"{A0C1853B-5C40-4B15-8766-3CF1C58F985A}~", ~"ProviderName~":~"Microsoft-Windows-PowerShell~", ~"EventId~":~"7937~", ~"EventData~":~"ContextInfo:         Severity = Informational
        Host Name = ConsoleHost
        Host Version = 5.1.22621.963
        Host ID = a5bbe5fe-ec7d-4202-a117-50e2b04ac735
        Host Application = C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
        Engine Version = 5.1.22621.963
        Runspace ID = 10b45360-8645-4c03-aee7-63635ebf557f
        Pipeline ID = 64
        Command Name = PSConsoleHostReadLine
        Command Type = Function
        Script Name = 
        Command Path = 
        Sequence Number = 302
        User = AFRICA\bosira
        Connected User = 
        Shell ID = Microsoft.PowerShell
 UserData:  Payload: Command PSConsoleHostReadLine is Stopped.
 ~"}
```

Reply:
===INPUT STRING

```
{~"TimeStamp~":~"2023-04-19T10:25:23.000Z~", ~"source~":~"ETW~", ~"Severity~":~"Information~", ~"ProviderId~":~"{A0C1853B-5C40-4B15-8766-3CF1C58F985A}~", ~"ProviderName~":~"Microsoft-Windows-PowerShell~", ~"EventId~":~"7937~", ~"**EventData**~":~"ContextInfo:         Severity = Informational
        Host Name = ConsoleHost
        Host Version = 5.1.22621.963
        Host ID = a5bbe5fe-ec7d-4202-a117-50e2b04ac735
        Host Application = C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe
        Engine Version = 5.1.22621.963
        Runspace ID = 10b45360-8645-4c03-aee7-63635ebf557f
        Pipeline ID = 64
        Command Name = PSConsoleHostReadLine
        Command Type = Function
        Script Name = 
        Command Path = 
        Sequence Number = 302
        User = AFRICA\bosira
        Connected User = 
        Shell ID = Microsoft.PowerShell
 UserData:  Payload: Command PSConsoleHostReadLine is Stopped.
 ~"}
```


===OUTPUT STRING

`{"TimeStamp":"2023-04-19T10:25:23.000Z", "source":"ETW", "Severity":"Information", "ProviderId":"{A0C1853B-5C40-4B15-8766-3CF1C58F985A}", "ProviderName":"Microsoft-Windows-PowerShell", "EventId":"7937", "**EventData**":"ContextInfo:         Severity = Informational\r\n        Host Name = ConsoleHost\r\n        Host Version = 5.1.22621.963\r\n        Host ID = a5bbe5fe-ec7d-4202-a117-50e2b04ac735\r\n        Host Application = C:\\WINDOWS\\System32\\WindowsPowerShell\\v1.0\\powershell.exe\r\n        Engine Version = 5.1.22621.963\r\n        Runspace ID = 10b45360-8645-4c03-aee7-63635ebf557f\r\n        Pipeline ID = 64\r\n        Command Name = PSConsoleHostReadLine\r\n        Command Type = Function\r\n        Script Name = \r\n        Command Path = \r\n        Sequence Number = 302\r\n        User = AFRICA\\bosira\r\n        Connected User = \r\n        Shell ID = Microsoft.PowerShell\r\n UserData:  Payload: Command PSConsoleHostReadLine is Stopped.\r\n "}`


Can we not pass the EventData section to the sanitize function as well? There are a bunch of \r and \n characters in there.

Reply:
@bobsira Please confirm if this is still occurring. This is the output from my side:

![image](https://user-images.githubusercontent.com/28774968/233344889-3ca51fe2-67af-457d-b2ca-21b24e63bca9.png)


`{"TimeStamp":"2023-04-20T10:51:53.000Z", "source":"ETW", "Severity":"Information", "ProviderId":"{DAA6A96B-F3E7-4D4D-A0D6-31A350E6A445}", "ProviderName":"Microsoft-Windows-WLAN-Driver", "EventId":"3", "EventData":"FrameUniqueID: 362284 CustomData1: 0 CustomData2: 0 CustomData3: 371 "}`

Reply:
Issue noted on some ETW providers like `Microsoft-Windows-PowerShell`.

The issue has also been noted in the `v2/main` branch as well and will be logged as a separate issue.